python/ray/serve/tests/test_metrics.py

```diff
@@ -88,6 +89,58 @@ def counter_log_success():
         wait_for_condition(counter_log_success)
 
 
+def test_http_metrics(serve_instance):
+    expected_metrics = [
+        "serve_num_http_requests", "serve_num_http_error_requests"
+    ]
+
+    def verify_metrics(expected_metrics, do_assert=False):
+        try:
+            resp = requests.get("http://127.0.0.1:9999").text
+        # Requests will fail if we are crashing the controller
+        except requests.ConnectionError:
+            return False
+        for metric in expected_metrics:
+            if do_assert:
+                assert metric in resp
+            if metric not in resp:
+                return False
+        return True
+
+    # Trigger HTTP 404 error
+    requests.get("http://127.0.0.1:8000/B/")
+    try:
+        wait_for_condition(
+            verify_metrics,
+            retry_interval_ms=1000,
+            timeout=10,
+            expected_metrics=expected_metrics)
+    except RuntimeError:
+        verify_metrics(expected_metrics, True)
+
+    expected_metrics.append("serve_num_deployment_http_error_requests")
+
+    @serve.deployment(name="A")
+    class A:
+        async def __init__(self):
+            pass
+
+        async def __call__(self):
+            # Trigger RayActorError
+            os._exit(0)
+
+    A.deploy()
+    requests.get("http://127.0.0.1:8000/A/")
+    try:
+        wait_for_condition(
+            verify_metrics,
+            retry_interval_ms=1000,
+            timeout=10,
+            expected_metrics=expected_metrics)
+    except RuntimeError:
+        verify_metrics(expected_metrics, True)
```

Comment:
Is it feasible to test that the number off errors is correct?  Also, is it feasible to test two deployments and check that the error count is tagged correctly per deployment?